subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstyleExtension.java

```diff
@@ -113,7 +116,18 @@ public File getConfigDir() {
      */
     @Incubating
     public void setConfigDir(File configDir) {
-        this.configDir = configDir;
+        this.configDir.set(configDir);
```

Comment:
I think we should `@Deprecated` these methods.

Reply:
I think the idea was to do that a little later: https://github.com/gradle/gradle/pull/4507#discussion_r173125501